src/v/cluster/persisted_stm.h

```diff
@@ -103,6 +103,12 @@ class persisted_stm
     ss::future<bool>
     wait_no_throw(model::offset offset, model::timeout_clock::duration);
 
+private:
+    ss::future<> wait_offset_committed(
+      model::timeout_clock::duration, model::offset, model::term_id);
+    ss::future<bool>
+      sync(model::timeout_clock::duration, model::offset, model::term_id);
```

Comment:
do you think we can name it `do_sync` ?, will be easier to differentiate with public API

Reply:
done